open-sphere-base/mantle/src/main/java/io/opensphere/mantle/plugin/selection/BufferRegionCreator.java

```diff
@@ -180,7 +183,26 @@ protected void destroyPreview()
     {
         if (myPreviewGeometry != null)
         {
-            unregisterGeometry(myPreviewGeometry);
+            if (myGeometry != null && myGeometry instanceof MultiPolygonGeometry)
+            {
+                MultiPolygonGeometry geometry = (MultiPolygonGeometry)myGeometry;
+                Collection<PolygonGeometry> geometries = geometry.getGeometries();
+                unregisterGeometry(myPreviewGeometry);
```

Comment:
since very branch does `unregisterGeometry(myPreviewGeometry);` can that just be done once at the beginning/end?

Reply:
not really.... because of how this is happening, if you un-register it before grabbing the geometries, they'll be lost before you can collect them. Likewise, adding them back only needs to happen if it's a group.